controllers/statussync/policy_status_sync.go

```diff
@@ -441,7 +490,225 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, request reconcile.Requ
 	return reconcile.Result{}, nil
 }
 
+func parseComplianceFromMessage(message string) policiesv1.ComplianceState {
+	cleanMsg := strings.ToLower(
+		strings.TrimSpace(
+			strings.TrimPrefix(message, "(combined from similar events):"),
+		),
+	)
+
+	if strings.HasPrefix(cleanMsg, "compliant") {
+		return policiesv1.Compliant
+	} else if strings.HasPrefix(cleanMsg, "pending") {
+		return policiesv1.Pending
+	}
+
+	return policiesv1.NonCompliant
+}
+
+// ceRequestFromEvent converts a Kubernetes event relating to a policy controller's compliance event to a
+// struct representing a POST request to compliance events API. An error is returned if the event can't be converted.
+func ceRequestFromEvent(event *corev1.Event) (utils.ComplianceAPIEventRequest, error) {
```

Comment:
Where handle `EventDetails.Metadata`

Reply:
The `metadata` is for additional details related to the compliance event (not the metadata section of the Kubernetes event). It's not utilized right now and is there for use in the future such as recording the noncompliant diff of a Kubernetes object.